artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/HierarchicalObjectRepository.java

```diff
@@ -477,50 +476,91 @@ default String modify(String input) {
 
    /**
     * Compares to matches to see which one is more specific.
+    * A match on the any-words delimiter (#) is considered less specific than a match without it, i.e. abc.def.# is less specific than abc.def and abc.def and abc.d*
```

Comment:
Either repetition or some other issue: "is less specific than abc.def and abc.def"